internal/provider/waypoint/resource_waypoint_application.go

```diff
@@ -50,6 +53,29 @@ type ApplicationResourceModel struct {
 
 	// deferred and probably a list or objects, but may possible be a separate
 	// ActionCfgs types.List `tfsdk:"action_cfgs"`
+
+	InputVars types.Set `tfsdk:"app_input_vars"`
+
+	// NOTE: At the time of writing this comment, TemplateInputVars is the only
+	// struct field that makes ApplicationResourceModel different from ApplicationDataSourceModel.
+	// One might see an opportunity here to use an embedded struct to avoid code duplication;
+	// however, this is not currently possible in the framework. See this issue for more details:
+	// https://github.com/hashicorp/terraform-plugin-framework/issues/242
+	TemplateInputVars types.Set `tfsdk:"template_input_vars"`
```

Comment:
Depending upon how [this issue](https://github.com/hashicorp/terraform-plugin-framework/issues/242) is addressed, we may be able to simplify this struct to embed `ApplicationDataSourceModel` into this one.